fdb-record-layer-lucene/src/test/java/com/apple/foundationdb/record/lucene/LuceneIndexTest.java

```diff
@@ -1167,6 +1186,151 @@ void testDataLoad() {
+            assertIndexEntryPrimaryKeys(Set.of(1000L, 1004L),  // 16,20
+                    recordStore.scanIndex(index, fullTextSearch(index, "four"), null, ScanProperties.FORWARD_SCAN));
+            assertIndexEntryPrimaryKeys(Set.of(),
+                    recordStore.scanIndex(index, fullTextSearch(index, "seven"), null, ScanProperties.FORWARD_SCAN));
+
+            if (primaryKeySegmentIndexEnabled) {
+                // TODO: Is there a more stable way to check this?
+                final LucenePrimaryKeySegmentIndex primaryKeySegmentIndex = ((LuceneIndexMaintainer)recordStore.getIndexMaintainer(index))
+                        .getDirectory(Tuple.from())
+                        .getPrimaryKeySegmentIndex();
+                assertEquals(List.of(
+                                List.of(1000L, "_p", 2),
+                                List.of(1001L, "_p", 0),
+                                List.of(1002L, "_n", 0),
+                                List.of(1003L, "_p", 3),
+                                List.of(1004L, "_q", 0)
+                        ),
+                        primaryKeySegmentIndex.readAllEntries());
+            }
+        }
+    }
+
+    @Test
+    void simpleDeleteSegmentIndex() {
+        final Index index = SIMPLE_TEXT_SUFFIXES_WITH_PRIMARY_KEY_SEGMENT_INDEX;
+        final RecordLayerPropertyStorage contextProps = RecordLayerPropertyStorage.newBuilder()
+                .addProp(LuceneRecordContextProperties.LUCENE_MERGE_SEGMENTS_PER_TIER, 3.0)
+                .build();
+        try (FDBRecordContext context = openContext()) {
+            rebuildIndexMetaData(context, SIMPLE_DOC, index);
+            recordStore.saveRecord(createSimpleDocument(1623L, ENGINEER_JOKE, 2));
+            recordStore.saveRecord(createSimpleDocument(1624L, ENGINEER_JOKE, 2));
+            context.commit();
+        }
+        try (FDBRecordContext context = openContext()) {
+            rebuildIndexMetaData(context, SIMPLE_DOC, index);
+            recordStore.saveRecord(createSimpleDocument(1547L, WAYLON, 2));
+            context.commit();
+        }
+        try (FDBRecordContext context = openContext()) {
+            rebuildIndexMetaData(context, SIMPLE_DOC, index);
+            assertIndexEntryPrimaryKeys(Set.of(1623L, 1624L),
+                    recordStore.scanIndex(index, fullTextSearch(index, "Vision"), null, ScanProperties.FORWARD_SCAN));
+            assertTrue(recordStore.deleteRecord(Tuple.from(1624L)));
+            assertIndexEntryPrimaryKeys(Set.of(1623L),
+                    recordStore.scanIndex(index, fullTextSearch(index, "Vision"), null, ScanProperties.FORWARD_SCAN));
+            assertThat(timer.getCount(LuceneEvents.Counts.LUCENE_DELETE_DOCUMENT_BY_QUERY), equalTo(0));
+            assertThat(timer.getCount(LuceneEvents.Counts.LUCENE_DELETE_DOCUMENT_BY_PRIMARY_KEY), equalTo(1));
+        }
+        timer.reset();
+        try (FDBRecordContext context = openContext()) {
```

Comment:
Given some of the concerns above, it seems like it would be good to have a test where you delete all the documents, and then commit, and re-open and check that PrimaryKeySegmentIndex is empty

Reply:
Added.